### PR TITLE
Fix the broken Istio installation command

### LINKED ISSUE #1338: Installing Istio 1.1.3 lean based on PR#1314 fails - istio-system namespace not created
## Expected Behavior
Istio Installs.

## Actual Behavior
istio-system namespace isn't created.
```
...
clusterrole.rbac.authorization.k8s.io "istio-pilot-istio-system" configured
clusterrole.rbac.authorization.k8s.io "istio-citadel-istio-system" configured
clusterrole.rbac.authorization.k8s.io "istio-sidecar-injector-istio-system" configured
clusterrole.rbac.authorization.k8s.io "istio-reader" configured
mutatingwebhookconfiguration.admissionregistration.k8s.io "istio-sidecar-injector" configured
Error from server (NotFound): error when creating "istio-lean.yaml": namespaces "istio-system" not found
Error from server (NotFound): error when creating "istio-lean.yaml": namespaces "istio-system" not found
Error from server (NotFound): error when creating "istio-lean.yaml": namespaces "istio-system" not found
Error from server (Invalid): error when creating "istio-lean.yaml": ClusterRoleBinding.rbac.authorization.k8s.io "istio-cleanup-secrets-istio-system" is invalid: subjects[0].kind: Unsupported value: "ServiceAccoun": supported values: "ServiceAccount", "User", "Group"
...
```
## Steps to Reproduce the Problem
1. Clean GKE system; created cluster-admin for me
1. Follow the instructions: https://github.com/knative/docs/blob/master/docs/install/installing-istio.md
1. Download Istio
```
export ISTIO_VERSION=1.1.3
curl -L https://git.io/getLatestIstio | sh -
cd istio-${ISTIO_VERSION}
```
1. Install CRD's
for i in install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl apply -f $i; done
1. Installing Istio without sidecar injection
```
helm template --namespace=istio-system \
  --set global.proxy.autoInject=disabled \
  --set global.omitSidecarInjectorConfigMap=true \
  --set global.disablePolicyChecks=true \
  --set prometheus.enabled=false \
  `# Disable mixer prometheus adapter to remove istio default metrics.` \
  --set mixer.adapters.prometheus.enabled=false \
  `# Disable mixer policy check, since in our template we set no policy.` \
  --set global.disablePolicyChecks=true \
  `# Set gateway pods to 1 to sidestep eventual consistency / readiness problems.` \
  --set gateways.istio-ingressgateway.autoscaleMin=1 \
  --set gateways.istio-ingressgateway.autoscaleMax=1 \
  `# Set pilot trace sampling to 100%` \
  --set pilot.traceSampling=100 \
  install/kubernetes/helm/istio \
  `# Removing trailing whitespaces to make automation happy` \
  | sed 's/[ \t]*$//' \
  > ./istio-lean.yaml

kubectl apply -f istio-lean.yaml
```

## Additional Info
**Additional context**

**Install information:**
 - Platform (GKE, IKS, AKS, etc.): GKE
 - Knative Version: haven't gotten that far, but was trying for 0.6.0

### PR DESCRIPTION
**Pull request by ZhiminXiang**
Fixes #1338

## Proposed Changes

- Add a step to create istio-system namespace
- Update the command of installing istio-lean.yaml according to the newest [download-istio.sh](https://github.com/knative/serving/blob/master/third_party/istio-1.1.3/download-istio.sh#L68)

### CODE REVIEW

**Comment by samodell** [comment]
Thanks, @ZhiminXiang ! I don't have time to test this before I hop on a plane, but I'll go ahead and approve it, so hopefully you can find someone to LGTM when it's been reviewed for technical accuracy.

/approve

File: docs/install/installing-istio.md [hunk 1]
```diff
@@ -84,24 +96,31 @@ mesh by [manually injecting the Istio sidecars][2].
 Enter the following command to install Istio:
 
 ```shell
-# A lighter template, with no sidecar injection.
+# A lighter template, with just pilot/gateway.
+# Based on install/kubernetes/helm/istio/values-istio-minimal.yaml
```
Comment (sebgoa) [inline docs/install/installing-istio.md]: I would expand that sentence a bit to " Based on the content of the ... directory"

**Comment by sebgoa** [comment]
I did not test but it seems good
/lgtm

File: docs/install/installing-istio.md [hunk 2]
```diff
@@ -69,6 +69,18 @@ without automatic sidecar injection.
...
+    ```shell
+    cat <<EOF | kubectl apply -f -
+    apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: istio-system
+      labels:
+        istio-injection: disabled
+    EOF
+    ```
```
Comment (lesv) [inline docs/install/installing-istio.md]: Right idea -- likely Ok, but when I copy & paste from this PR, zsh is unhappy w/ it.

**Comment by ZhiminXiang** [comment]
> After install, I see only istio-ingressgateway & istio-pilot pods, I then try to install knative 0.6 and I don't see any additional pods.
> 
> I'm going to try again on a new cluster.

Recently we had a change to strip down our lean install to just Pilot/Gateway https://github.com/knative/serving/pull/4076

Pilot and Gateway should be enough to run Knative (unless you need other features from Istio).

File: docs/install/installing-istio.md (hunk 2 above)
Comment (ZhiminXiang) [reply docs/install/installing-istio.md]: Could you please share any error information about this? This piece works well in my side.

**Comment by lesv** [comment]
The problem was that kNative didn't install (I should have been more clear in my comment).  I'm going to do everything from scratch and make sure I'm not doing something wrong, should be this afternoon.

File: docs/install/installing-istio.md (hunk 2 above)
Comment (lesv) [reply docs/install/installing-istio.md]: The leading spaces was causing failure.  It went into "pipe heredoc>" mode rather than ending.

Comment (ZhiminXiang) [reply docs/install/installing-istio.md]: I think if you copy it from the doc directly (https://github.com/knative/docs/blob/master/docs/install/installing-istio.md#downloading-istio-and-installing-crds), there should be leading spaces.

**Pull request merged**